app/services/PSRSubmissionService.scala

```diff
@@ -17,219 +17,44 @@
 package services
 
 import cats.implicits._
-import config.Refined.OneTo5000
 import connectors.PSRConnector
-import eu.timepit.refined.refineV
-import models.ConditionalYesNo._
 import models.SchemeId.Srn
-import models._
 import models.requests.DataRequest
 import models.requests.psr._
-import pages.nonsipp.{CheckReturnDatesPage, WhichTaxYearPage}
-import pages.nonsipp.common.{
-  CompanyRecipientCrnPage,
-  IdentityTypes,
-  OtherRecipientDetailsPage,
-  PartnershipRecipientUtrPage
-}
+import pages.nonsipp.CheckReturnDatesPage
 import pages.nonsipp.loansmadeoroutstanding._
-import pages.nonsipp.schemedesignatory.{
-  FeesCommissionsWagesSalariesPage,
-  HowManyMembersPage,
-  HowMuchCashPage,
-  ValueOfAssetsPage,
-  WhyNoBankAccountPage
-}
+import transformations.{LoanTransactionsTransformer, MinimalRequiredSubmissionTransformer}
 import uk.gov.hmrc.http.HeaderCarrier
 
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
-class PSRSubmissionService @Inject()(psrConnector: PSRConnector, schemeDateService: SchemeDateService) {
-
-  private type OptionalRecipientDetails = Option[(String, RecipientIdentityType, Option[Boolean], Option[String])]
+class PSRSubmissionService @Inject()(
+  psrConnector: PSRConnector,
+  minimalRequiredSubmissionTransformer: MinimalRequiredSubmissionTransformer,
+  loanTransactionsTransformer: LoanTransactionsTransformer
+) {
 
   def submitMinimalRequiredDetails(
     srn: Srn
   )(implicit hc: HeaderCarrier, ec: ExecutionContext, request: DataRequest[_]): Future[Option[Unit]] =
-    buildMinimalRequiredSubmission(srn).map(psrConnector.submitMinimalRequiredDetails(_)).sequence
+    minimalRequiredSubmissionTransformer.transform(srn).map(psrConnector.submitMinimalRequiredDetails(_)).sequence
```

Comment:
We would also need to remove psrConnector.submitMinimalRequiredDetails(_) from here and the BE and use  psrConnector.submitPsrDetails instead. Not sure if this is the right moment to do that or do it as a part of other ticket or create a bug. What is your opinion @cyeniceri ?

Reply:
submitMinimalRequiredDetails and related codes are removed. We are only using submitPsrDetails for all submissions including minimalRequiredSubmission